pkg/ingester/client/streaming.go

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package client
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/prometheus/prometheus/model/labels"
+
+	"github.com/grafana/mimir/pkg/util/limiter"
+)
+
+type StreamingSeries struct {
+	Labels  labels.Labels
+	Sources []StreamingSeriesSource
+}
+
+type StreamingSeriesSource struct {
+	StreamReader *SeriesChunksStreamReader
+	SeriesIndex  uint64
+}
+
+type SeriesChunksStreamReader struct {
```

Comment:
Some comments on these exported structs would be useful

Reply:
Resolved in f7f809a